common-test/src/main/java/io/strimzi/test/k8s/KubeCluster.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.test.k8s;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstraction for a Kubernetes cluster, for example {@code oc cluster up} or {@code minikube}.
+ */
+public interface KubeCluster {
+
+    /** Return true iff this kind of cluster installed on the local machine. */
+    boolean isAvailable();
+
+    /** Return true iff this kind of cluster is running on the local machine */
+    boolean isClusterUp();
+
+    /** Attempt to start a cluster */
+    void clusterUp();
+
+    /** Attempt to stop a cluster */
+    void clusterDown();
+
+    /** Return a default client for this kind of cluster. */
+    KubeClient defaultClient();
+
+    static KubeCluster bootstrap() {
+        Logger logger = LoggerFactory.getLogger(KubeCluster.class);
+        final boolean shouldStartCluster = System.getenv("CI") == null;
```

Comment:
Is this a duplication of what we have in `KubeClusterResource` ? Can we change the name to the env var `CI` naming it something clearer. i.e. `START_CLUSTER` or something like that ?